### PR TITLE
Redesign badge to support 'no stars'

### PR DESCRIPTION
**Pull request by kosdiamantis**
badges

### CODE REVIEW

File: src/plugins/badge.coffee [hunk 1]
```diff
@@ -584,6 +597,8 @@ class Badge
 
     stars
 
+  _displayStars: (rating) -> if rating is 0 then false else true
```
Comment (harrisred) [inline src/plugins/badge.coffee]: What if rating is 0.2? We should show any stars. Maybe we should bypass the `_ratingToStars `.

File: spec/plugins/badge_spec.coffee [hunk 2]
```diff
@@ -81,6 +81,16 @@ describe 'Badge', ->
...
 
+      context 'when rating is 0', ->
+        beforeEach (done) ->
+          setSaPlugins(@default_settings, rating: 0)
+
+          renderPlugin done, =>
+            @subject = window.parent.document.getElementById('sa-badge-floating-plugin')
+
+        it 'adds no stars class to container', ->
+          expect(@subject.className).to.match(/sa-badge-no-stars/)
```
Comment (harrisred) [inline spec/plugins/badge_spec.coffee]: We could use the `include` matcher. `include('sa-badge-no-stars')`

File: src/plugins/badge.coffee (hunk 1 above)
Comment (kosdiamantis) [inline src/plugins/badge.coffee]: This one is a business decision 😏

File: src/plugins/badge.coffee [hunk 3]
```diff
@@ -584,6 +584,8 @@ class Badge
 
     stars
 
+  _displayStars: (rating) -> rating isnt 0
```
Comment (dkart) [inline src/plugins/badge.coffee]: Consider renaming `_displayStars` to `_noStars` (returning the opposite value):

```coffeescript
_noStars: (rating) -> rating is 0
```
as the following snippets become more appropriate: 

```coffeescript
 _ratingToStars: (rating = 0, limit = 5) ->
  return [] if @_noStars(rating)
  ..
```

```coffeescript
..
@$embeddedBadgePlugin().className += ' sa-badge-no-stars' if @_noStars(rating)
..
```

File: spec/plugins/badge_spec.coffee [hunk 4]
```diff
@@ -228,8 +241,7 @@ describe 'Badge', ->
         expect(window.parent.document.getElementById('sa-badge-style')).to.exist
 
       it 'adds the plugin markup to the embedded badge container', ->
-        expect(window.parent.document.getElementById('sa-badge-embedded-header')).to.exist
-        expect(window.parent.document.getElementById('sa-badge-embedded-footer')).to.exist
+        expect(window.parent.document.getElementById('sa-badge-embedded-plugin').innerHTML).to.not.eq('')
```
Comment (harrisred) [inline spec/plugins/badge_spec.coffee]: Maybe `to.not.be.empty`

Comment (kosdiamantis) [inline spec/plugins/badge_spec.coffee]: 👌

File: src/plugins/badge.coffee (hunk 3 above)
Comment (kosdiamantis) [inline src/plugins/badge.coffee]: 👌 Changed

**Comment by dkart** [comment]
LGTM 👍

**Pull request merged**